packages/react-dom/src/shared/DOMProperty.js

```diff
@@ -146,6 +150,15 @@ export function shouldRemoveAttribute(
   propertyInfo: PropertyInfo | null,
   isCustomComponentTag: boolean,
 ): boolean {
+  if (enableFilterEmptyStringAttributesDOM) {
+    if (
+      propertyInfo !== null &&
+      propertyInfo.removeEmptyString === true &&
+      value === ''
+    ) {
+      return true;
+    }
+  }
```

Comment:
This should move below line 178 here into the `propertyInfo !== null` check so we only do that once.

However, we also probably should not filter this for isCustomComponentTag since custom elements could have all kinds of semantics where this matters. We only try to fix the built-ins.

Reply:
Oh, yeah. Duh.